### ISSUE TITLE
Ads not displaying in a real device

### ISSUE THREAD

**Issue by yanbaha** [admob-plus/admob-plus#123]
Ads are working on Android Studio but not in a real device.
I will be grateful if anyone can provide me with a working implementation code. It has been weeks that I am trying to solve this.
Thanks

**Comment by snow-drop** [comment]
Same here. I've waited for a week for the ads to be serve but nothing. I could see network requests in my AdMob account but no impressions are being counted as no ads are being displayed. The app is released as production instead of debug. Hopefully someone could help.
Thanks!

**Comment by 123dma** [comment]
Take a look 
https://groups.google.com/forum/#!category-topic/google-admob-ads-sdk/android/D_6InIUnUfU

Incase anyone is reading this that had the same issue I had.... simply doing what Liran Barsisa said fixed it... delete google play services data and your app data. I had noticed an issue where I changed my package name and stopped getting ads served on one of my test devices. Simply deleting both data fixed it. THANK YOU. Days spent lost on this.
